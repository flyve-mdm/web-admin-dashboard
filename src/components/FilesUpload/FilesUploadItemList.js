import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

class FilesUploadItemList extends PureComponent {
    render() {
        return (
            <div className='files-list' >
                <div className='files-list__content'>
                    <div className='files-list__item'>
                        <div className='icon files-list__item-icon'>
                            <span className='documentIcon'></span>
                        </div>
                        <div className='files-list__item-content-primary'>
                            <div className='files-list__content-text-primary'>{this.props.fileData.name}</div>
                            <div className='files-list__content-text-secondary'>{this.props.fileData.extension}</div>
                            <div className='files-list__content-text-secondary'>{this.props.fileData.sizeReadable}</div>
                        </div>
                        <div className='files-list__item-content-secondary'>
                            <div className='icon files-list__item-icon'>
                                <span className='deleteIcon' style={{ fontSize: '18px' }} onClick={this.props.onRemove}></span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

FilesUploadItemList.propTypes = {
    fileData: PropTypes.object.isRequired,
    onRemove: PropTypes.func.isRequired
}

export default FilesUploadItemList