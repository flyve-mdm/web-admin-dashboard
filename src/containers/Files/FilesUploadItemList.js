import React, { Component } from 'react'

export default class FilesUploadItemList extends Component {
    render() {
        return (
            <div className='files-list' >
                <div className='files-list-content'>
                    <div className='files-list-item'>
                        <div className='icon item-icon'>
                            <span className='documentIcon'></span>
                        </div>
                        <div className='item-content-primary'>
                            <div className='content-text-primary'>{this.props.fileData.name}</div>
                            <div className='content-text-secondary'>{this.props.fileData.extension}</div>
                            <div className='content-text-secondary'>{this.props.fileData.sizeReadable}</div>
                        </div>
                        <div className='item-content-secondary'>
                            <div className='icon item-icon'>
                                <span className='deleteIcon' style={{ fontSize: '18px' }} onClick={this.props.onRemove}></span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
