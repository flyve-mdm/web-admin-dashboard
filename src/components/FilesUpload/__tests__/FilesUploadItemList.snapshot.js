import React from 'react'
import renderer from 'react-test-renderer'

import FilesUploadItemList from "../FilesUploadItemList"

describe('FilesUploadItemList', () => {
    test('renders FilesUploadItemList', () => {
        const component = renderer.create(
            <FilesUploadItemList 
                fileData={{
                    name: 'test',
                    extension: 'png',
                    sizeReadable: '10 kb'
                }}
                onRemove={() => {}}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})