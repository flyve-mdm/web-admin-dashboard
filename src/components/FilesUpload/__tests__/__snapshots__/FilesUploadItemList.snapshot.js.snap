// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesUploadItemList renders FilesUploadItemList 1`] = `
<div
  className="files-list"
>
  <div
    className="files-list__content"
  >
    <div
      className="files-list__item"
    >
      <div
        className="icon files-list__item-icon"
      >
        <span
          className="documentIcon"
        />
      </div>
      <div
        className="files-list__item-content-primary"
      >
        <div
          className="files-list__content-text-primary"
        >
          test
        </div>
        <div
          className="files-list__content-text-secondary"
        >
          png
        </div>
        <div
          className="files-list__content-text-secondary"
        >
          10 kb
        </div>
      </div>
      <div
        className="files-list__item-content-secondary"
      >
        <div
          className="icon files-list__item-icon"
        >
          <span
            className="deleteIcon"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "18px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
